meson.build

```diff
@@ -192,6 +192,15 @@ if cc.compiles(malloc_attribute_test, name : 'malloc attribute with arguments')
   add_project_arguments('-DHAVE_MALLOC_EXTENDED_ATTRIBUTE', language: 'c')
 endif
 
+if cc.has_function('closefrom', prefix: '#define _GNU_SOURCE\n#include <unistd.h>')
+  add_project_arguments('-DHAVE_CLOSEFROM', language: 'c')
+endif
+if os == 'Linux' and \
```

Comment:
`_GNU_SOURCE` is not specific to Linux, so binding it to that OS is incorrect.  just probe the function all the time.

Reply:
🤔 I didn't? Here `unistd.h` is checked for a declaration of `closefrom` regardless of OS. It *is* necessary to define a source-level compatibility macro since the declaration of `closefrom` is inside an `#ifdef __USE_MISC` guard. I define `_GNU_SOURCE` here since that's what's already defined in the source code.

Reply:
yes, i commented on the line that's wrong -- checking the OS